metadata/metadata_test.go

```diff
@@ -207,4 +295,203 @@ var _ = Describe("Metadata", func() {
 			}))
 		})
 	})
+
+	When("CopyAnnotationsWithPrefixReplacement/CopyAnnotationsByPrefix is called", func() {
+		It("should return all the annotations matching the given prefix", func() {
+			podSource := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{"foo": "bar", "baz": "qux"},
+				},
+			}
+			podDest := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Annotations: map[string]string{"quux": "corge"},
+				},
+			}
+			CopyAnnotationsWithPrefixReplacement(podSource, podDest, "fo", "ba")
```

Comment:
Empty line before to separate from test resources creation.

Reply:
Updated, thanks!

Reply:
Missing the error check. `Expect(...).To(Succeed())`